### DISCUSSION TITLE
403 error when trying from Android

### DISCUSSION THREAD

**Discussion by copypasteearth** [weather-gov/api#552] [Q&A]
Hi I am trying to make an Android app that uses this, I am doing something like 

```
var url = "https://api.weather.gov/points/39.7456,-97.0892"
val jsonObjectRequest = object: JsonObjectRequest(Request.Method.GET, url, null,
                Response.Listener { response ->
                    binding.textView.text = "Response: %s".format(response.toString())
                },
                Response.ErrorListener { binding.textView.text = "That didn't work!" }
            )
            {
                @Throws(AuthFailureError::class)
                override fun getHeaders(): Map<String, String> {
                    val headers = HashMap<String, String>()
                    headers.put("Content-Type", "application/json")
                    headers.put("UserAgent", System.getProperty("http.agent"))
                    return headers
                }
            }
            queue.add(jsonObjectRequest)
```

but I keep getting 403 error

**Comment by scadergit** [comment]
403 typically means it was rejected by our CDN, likely from a missing user agent. Given you attempted to provide one, we need the CDN error code that is returned in the 403 body.

**Comment by copypasteearth** [comment]
@scadergit this is what I got so far

```
<HTML><HEAD>
    <TITLE>Access Denied</TITLE>
    </HEAD><BODY>
    <H1>Access Denied</H1>
     
    You don't have permission to access "http&#58;&#47;&#47;api&#46;weather&#46;gov&#47;points&#47;39&#46;7456&#44;&#45;97&#46;0892" on this server.<P>
    Reference&#32;&#35;18&#46;90112817&#46;1652053292&#46;1a2075ed
    </BODY>
    </HTML>
```

**Comment by copypasteearth** [comment]
2022-05-08 19:45:19.636 11554-11584/? E/ErrorHttp: {Cache-Control=max-age=0, Connection=close, Content-Length=308, Content-Type=text/html, Date=Sun, 08 May 2022 23:45:20 GMT, Expires=Sun, 08 May 2022 23:45:20 GMT, Mime-Version=1.0, Server=AkamaiGHost, Strict-Transport-Security=max-age=31536000 ; includeSubDomains ; preload, Vary=Accept,Feature-Flags,Accept-Language, X-Android-Received-Millis=1652053519623, X-Android-Response-Source=NETWORK 403, X-Android-Selected-Protocol=http/1.1, X-Android-Sent-Millis=1652053519599, X-Edge-Request-ID=1a2833f3}

**Comment by copypasteearth** [comment]
Access Denied
You don't have permission to access "http://api.weather.gov/points/39.7456,-97.0892" on this server.
Reference #18.90112817.1652053292.1a2075ed

**Comment by scadergit** [comment]
The problem is that the user agent is too generic and getting lumped into a security rule to thwart abuse. You will have better results if you set the user agent to a string that represents your app/contact, a few examples:
```
WeatherView/1.0 (weatherview.com)
WeatherView/1.0 (info@weatherview.com)
```
This also provides a means to contact you if for some reason your application triggers a security rule. (Eventually, this will be replaced by API keys, but that is not our current priority.)
